### ISSUE TITLE
want to see multiple lines of regression

### ISSUE THREAD

**Issue by fgmart** [isenseDev/rSENSE#2047]
if you have data sets organized into multiple groups, you can select exactly one group and then create a regression line.

then, you can select a single different group and make a regression line for that.

now, you can switch back and forth between views of either group, one at a time, each with its own regression line.

we want to be able to have both groups enabled, and see both lines simultaneously. 

e.g.: http://isenseproject.org/visualizations/467

also -- color of regression line should match color of points in corresponding data set.

**Comment by jaypoulz** [comment]
Best fit lines are currently not associated to groups by design for multiple reasons:

1. Colored lines become confusing when you switch to lines mode (which is default on Timeline)
2. Group selection is not mutually exclusive. That is to say, if there is an expectation for regression lines to be colored by group, which color do you choose for a regression line with multiple active groups?
3. Black stands out. (it's not amongst the series colors)
4. If you turn on multiple groups and see regression lines for individual groups, it would be confusing if there is not any indication that they are associated with different groups. It would also be inconsistent considering how regression lines are created (the current grouping is *assumed*).

In other words:
The only way to implement this feature is to break mutual exclusion of grouping.

**Comment by fgmart** [comment]
I have a feeling we're talking past each other.

Here are some views of the project I'm working on.

"Series" group enabled:

![screen shot 2015-03-10 at 3 15 17 pm](https://cloud.githubusercontent.com/assets/1038856/6583115/5d398544-c738-11e4-9b0a-4cb31bd5c05f.png)

"Parallel" group enabled:

![screen shot 2015-03-10 at 3 15 05 pm](https://cloud.githubusercontent.com/assets/1038856/6583123/6e8f158e-c738-11e4-8eea-f21bb0329df6.png)

Both groups enabled:

![screen shot 2015-03-10 at 3 16 47 pm](https://cloud.githubusercontent.com/assets/1038856/6583131/7dbedc10-c738-11e4-8968-e69a3cd412e9.png)

...and the regression lines go away.

I want to see both lines when I enable both groups.

I'm OK with the lines being black if it makes the conversation easier.

**Comment by jaypoulz** [comment]
I understand your request, and I am try to say that it is inconsistent with the way regressions currently work.

1. If you show both lines and they're both black, then the user makes a new regression, will the user realize that the reason the line is in the middle of both lines is because of the group selection at create time?

**If** yes -> there is no problem.

**Else** -> how do we solve that?
1. Add color to the regression line by group
``This is hard to implement because the color choice becomes ambiguous for lines with with multiple groups selected, also because line mode would be confusing for similarly colored lines.``

2. Other ideas?
``Undefined``

**Comment by fgmart** [comment]
Let's say we just keep the black and don't worry about colorization.

I think if a user has two (or more) groups selected, and makes a regression, then yes all data points should be included. And in this case yes, you would get a third line between the other two.

I don't see this as a problem.... that would be a correct regression...

So basically the workflow I imagine is:

* individually select a group
* make a regression line
* select a diff group
* make another line
* select both groups
* both groups' lines appear

Is this hard technically? Or are you primarily worried about users getting confused?

**Comment by jaypoulz** [comment]
My only concern is user confusion. My fear is that users will see all of the names in the table and wonder why they render differently despite being named the same. In the past, it was clear that regressions you could see belonged to currently group selection.

**Comment by kcarcia** [comment]
TESTERS: Play around with multiple lines  of regressions on vis. You can export Fred’s data from http://isenseproject.org/visualizations/467 and play with it as part of your testing.

**Comment by bdonald25** [comment]
PASSED testing on Chrome 42.0.2311.135 (mac).

**Comment by JKinsman** [comment]
PASSED testing on Chrome 42.0.2311.135 (Mac OSX 10.10.2).

**Comment by jdaigneau** [comment]
PASSED testing on Chrome 42.0.2311.90 (mac).

**Comment by jdaigneau** [comment]
PASSED testing on Chrome 42.0.2311.90 (mac).
